packages/block-editor/src/components/block-list/use-multi-selection.js

```diff
@@ -159,6 +159,20 @@ export default function useMultiSelection( ref ) {
 
 		if ( startClientId.current === clientId ) {
 			selectBlock( clientId );
+
+			if ( isFinal ) {
```

Comment:
So what exactly is this doing, and why does it fix? Is it something where the logic we've moved from `onSelectionEnd` is only intended to apply for single-block selections?

If so, I might suggest even some (useless, but clarifying) named variables, e.g.

```js
const isSingularSelection = startClientId.current === clientId;
if ( isSingularSelection ) {
```

And optionally some clarifying comments about how this particular logic is distinct for singular selections.